connman/src/timezone.c

```diff
@@ -226,18 +226,99 @@ static char *find_origin(void *src_map, struct stat *src_st,
 	return NULL;
 }
 
+static char *get_timezone_alpha2(const char *zone)
+{
+	GIOChannel *channel;
+	struct stat st;
+	char **tokens;
+	char *line;
+	char *alpha2 = NULL;
+	gsize len;
+	int fd;
+
+	if (!zone)
+		return NULL;
+
+	fd = open(USR_SHARE_ZONEINFO_MAP, O_RDONLY | O_CLOEXEC);
+	if (fd < 0) {
+		connman_warn("failed to open zoneinfo map %s",
+							USR_SHARE_ZONEINFO_MAP);
+		return NULL;
+	}
+
+	if (fstat(fd, &st) < 0) {
```

Comment:
given that 'st' is not used, what is the point with this fstat()?

Reply:
I think I just simply forgot to do this db56d888e1f90585888398b499571399d4ceb249

Reply:
so, regular files and symlinks to regular files are ok, but not sockets, pipes, etc - makes sense